app/models/user.rb

```diff
@@ -344,15 +344,12 @@ def destroy
     super
   end
 
-  %w[military_information payment].each do |emis_method|
-    define_method(emis_method) do
-      emis_model = instance_variable_get(:"@#{emis_method}")
-      return emis_model if emis_model.present?
-
-      emis_model = "EMISRedis::#{emis_method.camelize}".constantize.for_user(self)
-      instance_variable_set(:"@#{emis_method}", emis_model)
-      emis_model
-    end
+  def military_information
+    @military_information ||= if Flipper.enabled?(:military_information_vaprofile)
```

Comment:
If flipper is in `id_card_attributes`, there's no need for it here, right?

Reply:
Is that because we'd never hit the `if Flipper.enabled?` part of the conditional here?  🤔 in `id_card_attributes`, if the flipper is not enabled, we call `@user.military_information` which calls this method. Otherwise this method isn't called? @LindseySaari can you check me on this?

Reply:
It looks to me like the only other place is `form_profile.rb` and it has the similar flipper protection there (which only calls `user.military_information` if `military_information_vaprofile` flipper feature is false). Still seems like it would be more straightforward to remove flipper from `user.rb` and only have it in these files that decide where to call it

Reply:
I get that, but just in case we just want to keep this extra layer of protection so we're not caught off guard with some meta programming that uses this (or some other unknown). I'm _pretty_ sure you're right, Trevor, but there's no harm in leaving it for now. We also have a follow-up ticket to address whether the `military_information` method is needed in this file at all, so we'll come back to this later, once we know everything works as-is.